src/crowdin_client.h

```diff
@@ -80,49 +82,62 @@ class CrowdinClient
     struct ProjectListing
     {
         std::wstring name;
-        std::string identifier;
-        bool downloadable;
+        int id;
     };
 
     /// Retrieve listing of projects accessible to the user
     dispatch::future<std::vector<ProjectListing>> GetUserProjects();
 
+    // File information
+    struct FileInfo
+    {
+        std::wstring pathName;
+        int id, dirId, branchId;
+        
+    };
+
+
     /// Project detailed information
     struct ProjectInfo
     {
         std::wstring name;
-        std::string identifier;
+        int id;
         std::vector<Language> languages;
-        std::vector<std::wstring> files;
+        std::vector<FileInfo> files;
     };
 
     /// Retrieve listing of projects accessible to the user
-    dispatch::future<ProjectInfo> GetProjectInfo(const std::string& project_id);
+    dispatch::future<ProjectInfo> GetProjectInfo(const int project_id);
 
     /// Asynchronously download specific Crowdin file into @a output_file.
-    dispatch::future<void> DownloadFile(const std::string& project_id,
-                                        const std::wstring& file,
-                                        const Language& lang,
+    dispatch::future<void> DownloadFile(const long project_id,
+                                        const long file_id,
+                                        const wxString& file_name,
+                                        const std::string& lang_tag,
                                         const std::wstring& output_file);
 
     /// Asynchronously upload specific Crowdin file data.
-    dispatch::future<void> UploadFile(const std::string& project_id,
-                                      const std::wstring& file,
-                                      const Language& lang,
+    dispatch::future<void> UploadFile(const long project_id,
+                                      const long file_id,
+                                      const std::string& lang_tag,
```

Comment:
Just to pass to function only what is really used inside `UploadFile` while `Language` contains data which are no used there. Since personally me it's easier to read new code when I clearly see what is really needed inside particular function already by it arguments, not looking inside. Though I see and you point as well and if it's not affected by my explanation - I will change it back as before.

Reply:
> Just to pass to function only what is really used inside UploadFile while Language contains data which are no used there.

Consider the _user_ of this class -- i.e. other Poedit code that is read or written by people who don't necessarily breathe & live Crowdin implementation details. Is it reasonable to expect the users to have good understanding of what form of language tags does Crowdin use. Even I don't know off-hand and would have to look it up. If I were to bet, I would say BCP47, i.e. something different from what is used in PO files (locales), so there's room for confusion: locale is something you may automatically resolve to when in Poedit mindset, and that might be wrong.

If you _are_ aware of the potential complication, you'll look it up, e.g. by grepping crowdin_client.cpp or its uses and seeing what is done. But that still means that merely using the class from the outside is trickier, and relies on an implementation detail of the Crowdin API. And if the specific language tag format required changes in v8.0 of the API, client code using it would need to be changed, instead of just the implementation of this class.

By using Language in the API, the implementation detail of how Crowdin likes to ID languages is located only in its implementation and irrelevant to the rest of the code.